core/src/main/java/hudson/FilePath.java

```diff
@@ -2008,6 +2021,23 @@ public void copyTo(FilePath target) throws IOException, InterruptedException {
      * @since 1.311
      */
     public void copyToWithPermission(FilePath target) throws IOException, InterruptedException {
+        // Use NIO copy with StandardCopyOption.COPY_ATTRIBUTES when copying on the same machine.
+        if (this.channel == target.channel) {
+            act(new SecureFileCallable<Void>() {
+                public Void invoke(File f, VirtualChannel channel) throws IOException {
+                    try {
+                        File targetFile = new File(target.remote).getAbsoluteFile();
+                        mkdirs(targetFile.getParentFile());
```

Comment:
Should be `Files.createDirectory()`, otherwise you miss the directory creation failure event 🐛

Reply:
Yes good catch, I was originally going to update the implementation of `mkdirs(File)` in this PR to use `Files#createDirectories` but did not yet do so. I'll fix it.